Update nf-synchapi-initoncebegininitialize.md

AlexGuteniev opened the pull request:
Strengthen error safety

AlexGuteniev committed:
Update nf-synchapi-initoncebegininitialize.md

PRMerger8 commented:
@AlexGuteniev : Thanks for your contribution! The author(s) have been notified to review your proposed change.

Karl-Bridge-Microsoft committed:
Update nf-synchapi-initoncebegininitialize.md

The suggested edit doesn't improve clarity. The first note in Remarks addresses this qualification "can be used for either ... or ...".

I did clean up some of the md formatting.

Karl-Bridge-Microsoft commented:
#sign-off

AlexGuteniev commented:
@Karl-Bridge-Microsoft , I see the part I was proposing was removed:

> The function is not expected to fail (even under low resources condition) if it is used correctly and no attempts of concurrent synchronous and asynchronous initialization are made.

Was in confirmed to be incorrect?
